File: lib/utils/animations/entrance/elastic_drop.dart
import '../base/animated_text_state.dart';

AnimatedTextState evaluateElasticDrop(double t) {
  final dropOffset = -2.0 * (1.0 - t);
  final opacity = t < 0.1 ? t * 10.0 : 1.0;
  return AnimatedTextState(offsetY: dropOffset, opacity: opacity);
}

File: android/app/src/main/kotlin/com/typography/animations/entrance/ElasticDrop.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState
import com.typography.EasingType
import com.typography.animations.EasingFunctions

fun evaluateElasticDrop(t: Float): AnimatedTextState {
    val dropOffset = -2.0f * (1f - EasingFunctions.applyEasing(t, EasingType.ELASTIC_OUT))
    return AnimatedTextState(offsetY = dropOffset, opacity = if (t < 0.1f) t * 10f else 1f)
}
